src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { peopleFromServer } from '../data/people';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+type Props = {
+  onSelected?: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [list, setList] = useState<boolean>(false);
```

Comment:
The `list` state is defined as a boolean, which is correct, but consider renaming it to something more descriptive like `isListVisible` to improve code readability.